pkg/dynamicrp/integrationtest/dynamic/providers_test.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dynamic
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	v1 "github.com/radius-project/radius/pkg/armrpc/api/v1"
+	"github.com/radius-project/radius/pkg/dynamicrp/testhost"
+	"github.com/radius-project/radius/pkg/to"
+	"github.com/radius-project/radius/pkg/ucp/api/v20231001preview"
+	ucptesthost "github.com/radius-project/radius/pkg/ucp/testhost"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	radiusPlaneName           = "testing"
+	resourceProviderNamespace = "Applications.Test"
+	resourceTypeName          = "exampleResources"
+	locationName              = v1.LocationGlobal
+	apiVersion                = "2024-01-01"
+
+	resourceGroupName   = "test-group"
+	exampleResourceName = "my-example"
+
+	exampleResourcePlaneID = "/planes/radius/" + radiusPlaneName
+	exampleResourceGroupID = exampleResourcePlaneID + "/resourceGroups/test-group"
+
+	exampleResourceID  = exampleResourceGroupID + "/providers/Applications.Test/exampleResources/" + exampleResourceName
+	exampleResourceURL = exampleResourceID + "?api-version=" + apiVersion
+)
+
+// This test covers the lifecycle of a dynamic resource.
+func Test_Dynamic_Resource_Lifecycle(t *testing.T) {
+	_, ucp := testhost.Start(t)
+
+	// Setup a resource provider (Applications.Test/exampleResources)
+	createRadiusPlane(ucp)
+	createResourceProvider(ucp)
+	createResourceType(ucp)
+	createAPIVersion(ucp)
+	createLocation(ucp)
```

Comment:
Wouldn't the following look better and easier to read?

```suggestion
	// Setup a resource provider (Applications.Test/exampleResources)
	ucp.createRadiusPlane()
	ucp.createResourceProvider()
	ucp.createResourceType()
	ucp.createAPIVersion()
	ucp.createLocation()
```

Reply:
Sure, send a PR :)

Reply:
I'd love to share and de-dedupe these test helpers, but not part of this PR. I'm happy to do a follow up on this, because we could simplify a lot of code.